src/main/java/org/opensearch/knn/index/query/rescore/RescoreContext.java

```diff
@@ -25,6 +25,8 @@ public final class RescoreContext {
     @Builder.Default
     private float oversampleFactor = DEFAULT_OVERSAMPLE_FACTOR;
 
+    public static final RescoreContext DISABLED_RESCORE_CONTEXT = RescoreContext.builder().oversampleFactor(0).build();
```

Comment:
oversampling as 0 sounds a bit risky, any bug introduced with code changes might end up having k = 0. Should we just have a default value?

Reply:
thats a good point. Let me address this.